File: frontend/src/types/reports.ts
export type ReportType =
  | "interview_summary"
  | "weekly_progress"
  | "monthly_progress"
  | "resume_ats"
  | "roadmap"
  | "admin_export";

export interface ReportItem {
  id: string;
  report_type: string;
  status: string;
  title: string;
  content_type?: string | null;
  ready_at?: string | null;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
  has_file: boolean;
}

export interface ReportDetail extends ReportItem {
  payload?: Record<string, unknown> | null;
}

export interface NotificationItem {
  id: string;
  title: string;
  body: string;
  channel: string;
  status: string;
  payload?: Record<string, unknown> | null;
  read_at?: string | null;
  sent_at?: string | null;
  created_at: string;
}

File: backend/app/schemas/reports.py
"""Report request/response schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import ReportType


class CreateReportRequest(BaseModel):
    report_type: ReportType = ReportType.WEEKLY_PROGRESS
    title: str | None = Field(default=None, max_length=255)
    interview_id: UUID | None = None
    resume_id: UUID | None = None
    sync: bool = True
    format: Literal["json", "markdown", "pdf"] = "pdf"


class ReportResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    report_type: str
    status: str
    title: str
    content_type: str | None = None
    ready_at: datetime | None = None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime
    has_file: bool = False


class ReportDetailResponse(ReportResponse):
    payload: dict | None = None


class ReportAcceptedResponse(BaseModel):
    report_id: UUID
    status: str
    message: str
